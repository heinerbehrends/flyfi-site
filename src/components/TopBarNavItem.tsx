/** @jsx jsx */
import { jsx } from "theme-ui"
import React from "react"
import { Flex } from "theme-ui"
import { lighten, darken, saturation } from "@theme-ui/color"
import { Link } from "gatsby"

type TopBarNavItemProps = {
  linkText: string
  linkLocation: string
  variant?: "active"
}

export default function TopBarNavItem({
  linkText,
  linkLocation,
  variant,
}: TopBarNavItemProps) {
  return (
    <Link
      sx={{
        textDecoration: "none",
        color: `${variant === "active" ? "#fff" : "text"}`,
        fontWeight: variant === "active" ? 600 : 300,
      }}
      to={linkLocation}
    >
      <Flex
        sx={{
          height: "100%",
          backgroundColor: `${variant === "active" ? "muted" : "transparent"}`,
          " &:hover": { backgroundColor: "#a1e5e5" },
          alignItems: "center",
          paddingX: "3",
        }}
      >
        {linkText}
      </Flex>
    </Link>
  )
}
