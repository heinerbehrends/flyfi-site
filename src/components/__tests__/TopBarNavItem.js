import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import TopBarNavItem from "../TopBarNavItem";

describe("TopBarNavItem", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<TopBarNavItem linkText="home" linkLocation="/" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders the active variant correctly", () => {
    const variantTree = renderer
      .create(
        <TopBarNavItem linkText="home" linkLocation="/" variant="active" />
      )
      .toJSON();
    expect(variantTree).toMatchSnapshot();
  });
  it("renders a link with correct label and location", () => {
    render(<TopBarNavItem linkText="home" linkLocation="/" />);
    expect(screen.getByText("home")).toBeInTheDocument();
    expect(screen.getByRole("link")).toHaveAttribute("href", "/");
  });
});
