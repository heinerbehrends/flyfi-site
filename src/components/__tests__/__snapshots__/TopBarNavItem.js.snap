// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarNavItem renders correctly 1`] = `
.emotion-1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: text;
  font-weight: 300;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  background-color: transparent;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-0:hover {
  background-color: #a1e5e5;
}

<a
  className="emotion-1"
  href="/"
>
  <div
    className="emotion-0"
  >
    home
  </div>
</a>
`;

exports[`TopBarNavItem renders the active variant correctly 1`] = `
.emotion-1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  font-weight: 600;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  background-color: muted;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-0:hover {
  background-color: #a1e5e5;
}

<a
  className="emotion-1"
  href="/"
>
  <div
    className="emotion-0"
  >
    home
  </div>
</a>
`;
